httpd/handler.go

```diff
@@ -547,6 +551,24 @@ func (h *Handler) serveMetastore(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// serveShard returns a copy of the requested shard.
+func (h *Handler) serveShard(w http.ResponseWriter, r *http.Request) {
+	id := r.URL.Query().Get(":id")
+	shardID, err := strconv.ParseUint(id, 10, 64)
+	if err != nil {
+		httpError(w, "invalid shard ID", false, http.StatusBadRequest)
```

Comment:
We should send the parse error as well, so it shows the data that it tried to parse and err'd out on.